src/main.cpp

```diff
@@ -4867,6 +4873,157 @@ void static ProcessGetData(CNode* pfrom)
     }
 }
 
+bool checkVerificationQueue(){
+    return verificationQueue.isEmpty();
+}
+
+void txVerificationThread(){
+    int i = 0;
+    while(!ShutdownRequested())
+    {
+        if (!checkVerificationQueue() && i < 100)
+        {
+            LOCK(cs_verificationQueue);
+            verificationQueue.verifyOne();
+            i++;
+        }
+        else
+        {
+            i = 0;
+            std::this_thread::sleep_for (std::chrono::seconds(1));
+        }
+    }
+}
+
+void checkOneTx(CNode* pfrom, CTransaction& tx){
```

Comment:
tx could be const ref

Reply:
done
https://github.com/andreysobol/zen/commit/0493d81b4ddfad1832426ac1dcfe2630ee138fcb